### PR TITLE
Fix transitive dependencies errors in spago 0.20.0

### PR DESCRIPTION
**Pull request by milesfrain**
**Description of the change**

Fixes spago warnings when building and running tests for this library locally.

---

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a link to this PR and your username

### CODE REVIEW

**Comment by milesfrain** [comment]
Having trouble reproducing the CI failure locally. This command works fine for me:
```
spago build --purs-args '--censor-lib --strict --censor-codes="UserDefinedWarning"'
```
```
purs --version
0.14.1
spago --version
0.20.1
```

**Comment by thomashoneyman** [comment]
We can remove the unused function — that’s a 0.14.1 warning.

File: test/Examples.purs [hunk 1]
```diff
@@ -42,14 +42,14 @@ exampleContent1 =
...
+                                                                        {-
   let
     oneIfA = 1 <$ string "a" <?> "Letter was 'a'"
     zeroIfNotA = 0 <$ regex "[^a]" <?> "Letter was not 'a'"
     letterIsOneOrZero = oneIfA <|> zeroIfNotA <?>
                             "The impossible happened: \
                             \a letter was not 'a', and was not not-'a'."
     convertLettersToList = many1 letterIsOneOrZero
-                                                                        {-
   list <- convertLettersToList                                          -}
```
[the remark this answers is not in the document]
Comment (thomashoneyman) [reply test/Examples.purs:53]: I think it should be removed if it's unused.

[the remark this answers is not in the document]
Comment (JordanMartinez) [reply test/Examples.purs:53]: The point of this comment was to clarify what the below code is doing in a more piece-meal way.

**Pull request merged**